lib/yao/mode.rb

```diff
@@ -0,0 +1,19 @@
+module Yao
+  module Mode
+    def read_only!(&blk)
+      raise unless block_given?
+
+      Yao.config.set :raise_on_write, true
+      yield(blk)
```

Comment:
1. This statement passes `blk` as the first argument to `blk` itself. Is it the desired behaviour? 
2. What if `yield` raise? It looks `raise_on_write` is not correctly restored to false.